buster/docparser.py

```diff
@@ -84,25 +89,89 @@ def get_all_documents(
     return df
 
 
-def precompute_embeddings(df: pd.DataFrame) -> pd.DataFrame:
-    df["embedding"] = df.content.apply(lambda x: np.asarray(get_embedding(x, engine=EMBEDDING_MODEL), dtype=np.float32))
+def max_word_count(df: pd.DataFrame, max_words: int, col: str = "content") -> pd.DataFrame:
+    """Trim the word count of an entry to max_words"""
+    assert df[col].apply(lambda s: isinstance(s, str)).all(), "Column {col} must contain only strings"
+    word_counts_before = df[col].apply(lambda x: len(x.split(" ")))
+    df[col] = df[col].apply(lambda x: " ".join(x.split()[:max_words]))
+    word_counts_after = df[col].apply(lambda x: len(x.split(" ")))
+
+    trimmed = df[word_counts_before == word_counts_after]
+    logger.info(f"trimmed {len(trimmed)} documents to {max_words} words.")
+
     return df
 
 
-def generate_embeddings(root_dir: str, output_filepath: str, source: str) -> pd.DataFrame:
-    # Get all documents and precompute their embeddings
+def compute_embeddings(df: pd.DataFrame, engine: str = EMBEDDING_MODEL) -> pd.DataFrame:
+    logger.info(f"Computing embeddings for {len(df)} documents...")
+    df["embedding"] = df.content.apply(lambda x: np.asarray(get_embedding(x, engine=engine), dtype=np.float32))
+    logger.info(f"Done computing embeddings for {len(df)} documents.")
+    return df
+
+
+def generate_embeddings_parser(root_dir: str, output_filepath: str, source: str) -> pd.DataFrame:
     documents = get_all_documents(root_dir, supported_docs[source]["base_url"], supported_docs[source]["parser"])
+    return generate_embeddings(documents, output_filepath)
+
+
+def documents_to_db(documents: pd.DataFrame, output_filepath: str):
+    logger.info("Preparing database...")
+    sources = documents["source"].unique()
+    for source in sources:
+        documents_manager = get_documents_manager_from_extension(output_filepath)(output_filepath)
+        documents_manager.add(source, documents)
+    logger.info(f"Documents saved to: {output_filepath}")
+
+
+def generate_embeddings(
+    documents: pd.DataFrame,
+    output_filepath: str = "documents.db",
+    max_words=500,
+    embedding_engine: str = EMBEDDING_MODEL,
+) -> pd.DataFrame:
+    # check that we have the appropriate columns in our dataframe
+
+    assert set(required_cols := ["content", "title", "url"]).issubset(
+        set(documents.columns)
+    ), f"Your datafraem must contain {required_cols}."
+
+    # Get all documents and precompute their embeddings
+    documents = max_word_count(documents, max_words=max_words)
     documents = compute_n_tokens(documents)
-    documents = precompute_embeddings(documents)
+    documents = compute_embeddings(documents, engine=embedding_engine)
 
-    documents_manager = get_documents_manager_from_extension(output_filepath)(output_filepath)
-    documents_manager.add(source, documents)
+    # save the documents to a db for later use
+    documents_to_db(documents, output_filepath)
 
     return documents
+
+
+@click.command()
```

Comment:
didn't know about click, it looks nice

Reply:
yes, alternative to argparse, simplifies the syntax or cli args